modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionListener.java

```diff
@@ -302,30 +256,14 @@ private void handleTxCleanupCommand(TxCleanupCommand cmd, long commandIndex) {
         storage.runConsistently(() -> {
             UUID txId = cmd.txId();
 
-            Set<ByteBuffer> removedKeys = txsRemovedKeys.getOrDefault(txId, Collections.emptySet());
-
-            Set<ByteBuffer> insertedKeys = txsInsertedKeys.getOrDefault(txId, Collections.emptySet());
-
             Set<RowId> pendingRowIds = txsPendingRowIds.getOrDefault(txId, Collections.emptySet());
 
             if (cmd.commit()) {
                 pendingRowIds.forEach(rowId -> storage.commitWrite(rowId, cmd.commitTimestamp()));
             } else {
-                pendingRowIds.forEach(rowId -> storage.abortWrite(rowId));
+                pendingRowIds.forEach(storage::abortWrite);
```

Comment:
Seems that index cleanup on rollback is missing.

Reply:
Yes, this patch is already to big, so I decided to slit clean up part to https://issues.apache.org/jira/browse/IGNITE-17974, if you don't mind